platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py

```diff
@@ -1457,35 +1459,36 @@ def get_tx_disable_channel(self):
         return tx_disabled
 
 
+    def mgmt_phy_mod_pwr_attr_get(self, power_attr_type):
+        sx_mgmt_phy_mod_pwr_attr_p = new_sx_mgmt_phy_mod_pwr_attr_t_p()
+        sx_mgmt_phy_mod_pwr_attr = sx_mgmt_phy_mod_pwr_attr_t()
+        sx_mgmt_phy_mod_pwr_attr.power_attr_type = power_attr_type
+        sx_mgmt_phy_mod_pwr_attr_t_p_assign(sx_mgmt_phy_mod_pwr_attr_p, sx_mgmt_phy_mod_pwr_attr)
+        try:
+            rc = sx_mgmt_phy_mod_pwr_attr_get(self.sdk_handle, self.sdk_index, sx_mgmt_phy_mod_pwr_attr_p)
+            assert SX_STATUS_SUCCESS == rc, "sx_mgmt_phy_mod_pwr_attr_get failed"
+            sx_mgmt_phy_mod_pwr_attr = sx_mgmt_phy_mod_pwr_attr_t_p_value(sx_mgmt_phy_mod_pwr_attr_p)
+            pwr_mode_attr = sx_mgmt_phy_mod_pwr_attr.pwr_mode_attr
+            return pwr_mode_attr.admin_pwr_mode_e, pwr_mode_attr.oper_pwr_mode_e
+        finally:
+            delete_sx_mgmt_phy_mod_pwr_attr_t_p(sx_mgmt_phy_mod_pwr_attr_p)
+
+
     def get_lpmode(self):
         """
         Retrieves the lpmode (low power mode) status of this SFP
 
         Returns:
             A Boolean, True if lpmode is enabled, False if disabled
         """
-        if self.sfp_type == QSFP_TYPE:
-            if self._open_sdk():
-                # Get MCION
-                mcion = ku_mcion_reg()
-                mcion.module = self.sdk_index
-                meta = self._init_sx_meta_data()
-                meta.access_cmd = SXD_ACCESS_CMD_GET
-
-                rc = sxd_access_reg_mcion(mcion, meta, REGISTER_NUM, None, None)
-                self._close_sdk()
-
-                if rc != SXD_STATUS_SUCCESS:
-                    logger.log_warning("sxd_access_reg_mcion getting failed, rc = %d" % rc)
-                    return None
+        handle = self._open_sdk()
+        if handle is None:
+            return False
 
-                # Get low power mode status
-                lpm_mask = 1 << 8
-                lpm_status = (lpm_mask & mcion.module_status_bits) != 0
+        admin_pwr_mode, oper_pwr_mode = self.mgmt_phy_mod_pwr_attr_get(SX_MGMT_PHY_MOD_PWR_ATTR_PWR_MODE_E)
 
-                return lpm_status
-        else:
-            return NotImplementedError
+        self._close_sdk()
```

Comment:
If it throw exception in self.mgmt_phy_mod_pwr_attr_get, it won't be able to close the SDK, will it leak any resource on SDK side?

Reply:
will address this in another PR to optimize the SDK handler usage, all the SFP object should share one handler.